commons/serialization/src/main/scala/sbt/pickling/CustomPicklerUnpickler.scala

```diff
@@ -72,26 +73,19 @@ trait CustomPicklerUnpickler extends LowPriorityCustomPicklerUnpickler {
     val tag = implicitly[FastTypeTag[Option[A]]]
     private val isPrimitive = elemTag.tpe.isEffectivelyPrimitive
     private val nullPickler = implicitly[SPickler[Null]]
-
+    private val nullTag = implicitly[FastTypeTag[Null]]
     def pickle(coll: Option[A], builder: PBuilder): Unit = {
-      builder.beginEntry(coll)
-      builder.pushHints()
-
+      // Here we cheat the "entry" so that the notion of option
+      // is erased for "null"
       coll match {
         case Some(elem) =>
-          if (isPrimitive) {
-            builder.hintStaticallyElidedType()
```

Comment:
Yes.  I had forgotten since we decided to always emit a type even if we ignore it later